src/goto-programs/interpreter_evaluate.cpp

```diff
@@ -339,11 +339,19 @@ void interpretert::evaluate(
         evaluate(expr.op0(), dest);
         return;
       }
-      if(expr.has_operands() && !to_integer(expr.op0(), i))
+      else if(
+        expr.has_operands() && !to_integer(to_constant_expr(expr.op0()), i))
       {
         dest.push_back(i);
         return;
       }
+      // check if expression is constant null pointer without operands
+      else if(
+        !expr.has_operands() && !to_integer(to_constant_expr(expr), i) &&
+        i == 0)
```

Comment:
You might just do `expr.is_zero()`, although `config.ansi_c.NULL_is_zero` should actually be considered.